File: backend/src/main/java/com/inventory/backend/repository/PrinterSlotRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.PrinterSlot;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PrinterSlotRepository extends JpaRepository<PrinterSlot, Long> {
    List<PrinterSlot> findByPrinterIdOrderByIdAsc(Long printerId);
    long countByCartridgeModelId(Long cartridgeModelId);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/PrinterSlotRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.PrinterSlot
import org.springframework.data.jpa.repository.JpaRepository

interface PrinterSlotRepository : JpaRepository<PrinterSlot, Long> {
    fun findByPrinterIdOrderByIdAsc(printerId: Long): List<PrinterSlot>
    fun countByCartridgeModelId(cartridgeModelId: Long): Long
}
